llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -476,11 +476,46 @@ Error InstrProfSymtab::create(Module &M, bool InLTO) {
       return E;
   }
 
+  SmallVector<MDNode *, 2> Types;
+  for (GlobalVariable &G : M.globals()) {
+    if (!G.hasName() || !G.hasMetadata(LLVMContext::MD_type))
+      continue;
+    if (Error E = addVTableWithName(
+              G, getIRPGOObjectName(G, InLTO, /* PGONameMetadata */ nullptr)))
+        return E;
+  }
+
   Sorted = false;
   finalizeSymtab();
   return Error::success();
 }
 
+Error InstrProfSymtab::addVTableWithName(GlobalVariable &VTable,
+                                         StringRef VTablePGOName) {
+
+  auto mapName = [&](StringRef Name) -> Error {
+    // Use 'addSymbolName' rather than 'addVTableName' as 'VTableNames' is
```

Comment:
Perhaps the 'addVTableName' should be renamed to 'addVTableNameForWrite'?

Can the lambda be folded into addSymbolName method ? The method can take an additional bool parameter indicating if MD5VTableMap needs to be updated.

Reply:
AddFuncName also has a mapName lambda, perhaps it can be unified too by adding an enum parameter to addSymbolName(StringRef Name, InstProfSymtab::SymKind ) {
..}
SymKind { SK_Func, SK_Vtab, SK_Other};

Reply:
`MD5VTableMap` and `MD5FuncMap` have different container types (`DenseMap` vs `vector`), which makes hard to unify the container methods (`try_emplace` vs `emplace_back`) in a lambda function.

I added a TODO (in InstrProf.h) to unify the container type (but aim to use separate container instances), which would help to unify the lambda function.